tests/rptest/tests/tx_admin_api_test.py

```diff
@@ -209,3 +209,41 @@ def test_mark_transaction_expired(self):
                     assert (self.extract_pid(tx) in expected_pids)
                     assert (tx['status'] == 'ongoing')
                     assert (tx['timeout_ms'] == 60000)
+
+    @cluster(num_nodes=3)
+    def test_all_ttransactions(self):
+        tx_id = "0"
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+            'transactional.id': tx_id,
+        })
+        producer.init_transactions()
+        producer.begin_transaction()
+
+        for topic in self.topics:
+            for partition in range(topic.partition_count):
+                producer.produce(topic.name, '0', '0', partition)
+
+        producer.flush()
+
+        txs_info = self.admin.get_all_transactions()
+        assert (len(txs_info) == 1)
+
+        expected_partitions = dict()
+        tx = txs_info[0]
+
+        assert (tx["transactional_id"] == tx_id)
+        assert (tx["timeout_ms"] == 60000)
+
+        for partition in tx["partitions"]:
+            assert (partition["ns"] == "kafka")
+            if partition["topic"] not in expected_partitions:
+                expected_partitions[partition["topic"]] = set()
+            expected_partitions[partition["topic"]].add(
+                partition["partition_id"])
+
+        for topic in self.topics:
+            assert (len(
+                expected_partitions[str(topic)]) == topic.partition_count)
```

Comment:
why do you need `str(topic)`? if topic is a TopicSpec you can use `topic.name`

Reply:
I do not know. I like this syntax

Reply:
Please use `topic.name`. The `str(topic)` invokes the `__str__` method of the TopicSpec type which returns a human readable representation of a TopicSpec instance. While it currently returns `topic.name`, that is not a stable representation to depend on if you expect _exactly_ the topic name to be returned.